src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -591,11 +603,11 @@ object AppDefinition extends GeneralPurposeCombinators {
   def residentUpdateIsValid(from: AppDefinition): Validator[AppDefinition] = {
     val changeNoVolumes =
       isTrue[AppDefinition]("Persistent volumes can not be changed!") { to =>
-        val fromVolumes = from.persistentVolumes
-        val toVolumes = to.persistentVolumes
+        val fromVolumes = from.persistentVolumesAndMounts
+        val toVolumes = to.persistentVolumesAndMounts
```

Comment:
Are we checking more data than before? (mounts?) We should be able to allow the user to change mounts. This does not result in a reservation change.

Reply:
We are checking the same data as before.